signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java

```diff
@@ -184,6 +184,10 @@ public VerifyChallengeResponse verifyChallenge(VerifyChallengeRequest verifyChal
 
     public RegisterResponse register(RegisterRequest registerRequest, String transactionId) throws SignUpException {
         RegistrationTransaction transaction = cacheUtilService.getChallengeVerifiedTransaction(transactionId);
+        if(registerRequest.getUsername()!= registerRequest.getUserInfo().get("phone").asText()) {
```

Comment:
@balaji-alluru we cannot hardcode the "phone" field as we need to be dynamic in the schema support. Can we do this check in the ProfileRegistryPlugin. And lets do this check in the plugin based on the configured field Id instead of using "phone" directly. if nothing is configured then we can skip this check in the plugin.

Reply:
Also why is "!=" operator used for content comparison? kindly use "equals" method.

Reply:
updated the same anusha